### PR TITLE
Remove prediction/eval from preprocessor

### PR DESCRIPTION
**Pull request by waprin**
@nkashy1 if you want to take a peek where I'm at...kind of a mess but will hopefully be less messy soon.

### CODE REVIEW

File: spark-tensorflow/prepare/src/main/scala/com/google/cloud/ml/samples/criteo/CriteoFeatures.scala [hunk 1]
```diff
@@ -21,34 +21,22 @@ import org.apache.spark.sql.types.{StringType, StructField, StructType}
...
   val integerFeatureLabels: Seq[String] = (1 to 13).map(index => s"integer-feature-$index")
   val categoricalFeatureLabels: Seq[String] = (1 to 26).map(index => s"categorical-feature-$index")
   val categoricalRawLabels: Seq[String] = categoricalFeatureLabels.map({label => label + "-raw"})
   val clickedLabel = Seq("clicked")
 
-  val inputLabels: Seq[String] = mode match {
-    case Predict => integerFeatureLabels ++ categoricalRawLabels
-    case _ => clickedLabel ++ integerFeatureLabels ++ categoricalRawLabels
-  }
+  val inputLabels: Seq[String] = clickedLabel ++ integerFeatureLabels ++ categoricalRawLabels
```
Comment (nkashy1) [inline spark-tensorflow/prepare/src/main/scala/com/google/cloud/ml/samples/criteo/CriteoFeatures.scala]: Minor point: this should probably get a better name now?

File: spark-tensorflow/prepare/src/main/scala/com/google/cloud/ml/samples/criteo/CriteoIndexer.scala [hunk 2]
```diff
@@ -68,15 +68,13 @@ trait CriteoIndexer {
     }).
       toMap
 
-  def transform(rawDf: DataFrame, resource: IndexerResource): DataFrame = {
+  def getEmbeddings(resource: IndexerResource): Map[String, DataFrame] = {
```
Comment (nkashy1) [inline spark-tensorflow/prepare/src/main/scala/com/google/cloud/ml/samples/criteo/CriteoIndexer.scala]: As a solution to our problems with the term "embedding", let's just call them vocabularies (or ordered vocabularies)? That's consistent with the tf.feature_column names.

File: spark-tensorflow/prepare/src/main/scala/com/google/cloud/ml/samples/criteo/CriteoMissingReplacer.scala [hunk 3]
```diff
@@ -30,39 +30,45 @@ class CriteoMissingReplacer(val artifactExporter: ArtifactExporter)
                            (implicit val spark: SparkSession) {
```
[the remark this answers is not in the document]
Comment (nkashy1) [reply spark-tensorflow/prepare/src/main/scala/com/google/cloud/ml/samples/criteo/CriteoMissingReplacer.scala]: Follow-up: Looks like `artifactExporter.export` is being called in the `CriteoAnalyzer`, which means you don't have to inject it into `CriteoMissingReplacer`.

File: spark-tensorflow/prepare/src/main/scala/com/google/cloud/ml/samples/criteo/CriteoMissingReplacer.scala [hunk 4]
```diff
@@ -20,49 +20,53 @@ import org.apache.spark.sql._
...
+    * @return A map from integer column names to their averages
+    */
+  def getAverageIntegerFeatures(to_average_df: DataFrame,
+                                features: Seq[String]): Map[String, DataFrame] = {
     val integerFeaturesDf = to_average_df.
       select(features.head, features.tail: _*).
       toDF
 
-    val averages = integerFeaturesDf.
+    integerFeaturesDf.
```
Comment (nkashy1) [inline spark-tensorflow/prepare/src/main/scala/com/google/cloud/ml/samples/criteo/CriteoMissingReplacer.scala]: How are null entries treated by this?

One thing that we should be careful about is that the average for a given feature is only taken over cells in the DataFrame which do not have nulls. For this purpose, replacing nulls with 0 would not work.

Comment (waprin) [reply spark-tensorflow/prepare/src/main/scala/com/google/cloud/ml/samples/criteo/CriteoMissingReplacer.scala]: Spark sql average functions just ignores nulls so it does what we want

File: spark-tensorflow/prepare/src/main/scala/com/google/cloud/ml/samples/criteo/CriteoIndexer.scala (hunk 2 above)
Comment (waprin) [reply spark-tensorflow/prepare/src/main/scala/com/google/cloud/ml/samples/criteo/CriteoIndexer.scala]: done...i renamed embeddings to vocabularies and vocabularies to featurevaluecounts, that made sense to me, LMK what you think.

File: spark-tensorflow/prepare/src/main/scala/com/google/cloud/ml/samples/criteo/CriteoMissingReplacer.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (waprin) [reply spark-tensorflow/prepare/src/main/scala/com/google/cloud/ml/samples/criteo/CriteoMissingReplacer.scala]: Yeah could do it either way, seemed better to me to keep classes to narrow responsibilities.

**Pull request closed without merging**